File: leetcode/easy/2824.count-pairs-whose-sum-is-less-than-target.py
#
# @lc app=leetcode id=2824 lang=python3
#
# [2824] Count Pairs Whose Sum is Less than Target
#

# @lc code=start
class Solution:
    def countPairs(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums) - 1

        nums.sort()
        count = 0

        while (i<j):
            if ((nums[i] + nums[j]) < target):
                count += j - i
                i = i + 1
            else:
                j = j - 1
        
        return count
# @lc code=end

File: leetcode/2824.count-pairs-whose-sum-is-less-than-target.cpp
/*
 * @lc app=leetcode id=2824 lang=cpp
 *
 * [2824] Count Pairs Whose Sum is Less than Target
 */

// @lc code=start
class Solution {
public:
    int countPairs(vector<int>& nums, int target) {
        int i = 0, j = nums.size() - 1;
        sort(nums.begin(), nums.end());

        int count  = 0;
        while (i<j) {
            if ((nums[i] + nums[j]) < target) {
                count += j - i;
                i++;
            } else {
                j--;
            }
         }

         return count;
        
    }
};
// @lc code=end

